kolibri/plugins/management/assets/src/vue/user-page/index.vue

```diff
@@ -198,10 +202,29 @@
     },
     $trNameSpace: 'userPage',
     $trs: {
-      noUsersExist: 'No Users Exist.',
-      allUsersFilteredOut: 'No users match the filter.',
+      // input & accessibility labels
```

Comment:
> the main thing I don't understand is how translators are supposed to fully understand the context of what they're translating. 

Oh, I'm so glad somebody said this... :heart_eyes: 
So many times it's a guessing game, and it often makes the job of correctly localizing UI really challenging. 
> Would be optimal if somehow they could see a screenshot of the page with the text highlighted

That would be a big help, but not many CAT tools offer that option, and Crowdin certainly does not, afaik. This is what the translator sees (old `json` file for Management page, just two strings):

![management_module-messages json - kolibri - crowdin translation](https://cloud.githubusercontent.com/assets/1457929/22159529/a87320ba-df41-11e6-8b23-035f9c0c0423.png)

So `$trNameSpace` and `$tr` are indeed the only context available, and I guess we'll have to find a balance between verbosity and adequate amount of information. Maybe Devon's previous suggestion went a bit over the top, but we can start making a small list of conventions for the most common UI elements as the first part of the `trs`. We would document them in the dev docs and keep updating it as we integrate new features. Majority of translators would have no problem with acronyms like:

- placeholder-msg
- btn-lbl
- dropdown-sel-lbl
- dropdown-sel-opt

```
$trs: {
  'placeholder-msg.noUser': 'No users exist.',
  'placeholder-msg.allUsersFiltered': 'No users match the filter.',
  'btn-lbl.editUser': 'Edit account information',
  'dropdown-sel-lbl.filterUserType': 'Filter User Type',
  'dropdown-sel-opt.admins': 'Admins',
```

As for **DRY-ing** for translator's sake, not really necessary as when something is translated once, it remains in Kolibri's TM that is available as clickable suggestions below the string, with a match %, that can be easily reused, together with suggestions from other sources.